crates/csskit_derives/src/to_cursors.rs

```diff
@@ -0,0 +1,73 @@
+use proc_macro2::{Span, TokenStream};
+use quote::{format_ident, quote};
+use syn::{Data, DataEnum, DataStruct, DeriveInput, Fields, Index};
+
+use crate::err;
+
+pub fn derive(input: DeriveInput) -> TokenStream {
+	let ident = input.ident;
+	let generics = &mut input.generics.clone();
+	let (impl_generics, _, _) = generics.split_for_impl();
+	let body = match input.data {
+		Data::Struct(DataStruct { fields: Fields::Unnamed(fields), .. }) => {
+			let steps: Vec<TokenStream> = fields
+				.unnamed
+				.into_iter()
+				.enumerate()
+				.map(|(i, _)| {
+					let index = Index { index: i as u32, span: Span::call_site() };
+					quote! {
+						::css_parse::ToCursors::to_cursors(&self.#index, s);
+					}
+				})
+				.collect();
+			quote! { #(#steps)* }
+		}
+
+		Data::Struct(_) => err(ident.span(), "Cannot derive ToCursors on a struct with named fields"),
+
+		Data::Union(_) => err(ident.span(), "Cannot derive ToCursors on a Union"),
+
+		Data::Enum(DataEnum { variants, .. }) => {
```

Comment:
Would be cool to have one `assert_snapshot!` test, makes seeing what this compiled code looks like, much easier.